src/openrct2/PlatformEnvironment.h

```diff
@@ -10,24 +10,28 @@
 #pragma once
 
 #include "common.h"
+#include "util/Enum.hpp"
 
 #include <memory>
 #include <string>
 
 namespace OpenRCT2
 {
-    enum class DIRBASE : size_t
+    struct DIRBASE
     {
-        RCT1,          // Base directory for original RollerCoaster Tycoon 1 content.
-        RCT2,          // Base directory for original RollerCoaster Tycoon 2 content.
-        OPENRCT2,      // Base directory for OpenRCT2 installation.
-        USER,          // Base directory for OpenRCT2 user content.
-        CONFIG,        // Base directory for OpenRCT2 configuration.
-        CACHE,         // Base directory for OpenRCT2 cache files.
-        DOCUMENTATION, // Base directory for OpenRCT2 doc files.
+        using Type = IndexEnum<DIRBASE, size_t>;
+
+        static constexpr Type Init = { 0 };
+        static constexpr Type RCT1 = { 0 };          // Base directory for original RollerCoaster Tycoon 1 content.
+        static constexpr Type RCT2 = { 1 };          // Base directory for original RollerCoaster Tycoon 2 content.
```

Comment:
I think one disadvantage of this approach is having to manually define all enumeration values, which we do lots of times, but not always.

it also allows overlapping values, which is a feature in this specific part, but might hide a bug in another